homeassistant/components/google_assistant/smart_home.py

```diff
@@ -136,6 +135,23 @@ def sync_serialize(self):
         room = entity_config.get(CONF_ROOM_HINT)
         if room:
             device['roomHint'] = room
+        else:
+            entity_registry = (await self.hass.helpers.
+                               entity_registry.async_get_registry())
+
+            entity_entry = entity_registry.async_get(state.entity_id)
+            if entity_entry:
+                device_registry = (await self.hass.helpers.
+                                   device_registry.async_get_registry())
+                device_entry = device_registry.devices.get(
+                    entity_entry.device_id)
+                if device_entry and device_entry.area_id:
+                    area_registry = (await self.hass.helpers.
```

Comment:
I wonder if we should get all registries at once, to avoid having to yield 3 times. 

```python
dev_reg, ent_reg, area_reg = asyncio.gather(
    self.hass.helpers.device_registry.async_get_registry(),
    self.hass.helpers.entity_registry.async_get_registry(),
    self.hass.helpers.area_registry.async_get_registry(),
)
```

Reply:
Actually, we should move getting the registries into `async_devices_sync` so that we only have to get the registries once. Or maybe they should become part of `config` ?

Reply:
Let's start with just putting it in `async_devices_sync`

Reply:
I have pushed an update that changes all this. 

I was thinking however that it might be useful to have some helper methods:

in ```DeviceRegistry```:

```async def get_by_entity_id(entity_id) -> Optional[DeviceEntry]: ```

and in ```AreaRegistry```:

```async def get_by_entity_id(entity_id)  -> Optional[AreaEntry]: ```
```async def get_by_device_id(device_id) -> Optional[AreaEntry]: ```